#!/usr/bin/env python
PACKAGE = "terabee"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

undistortion_type = gen.enum([ gen.const("NONE",     int_t, 1, "undistortion will not be applied"),
                               gen.const("DEPTH",    int_t, 2, "undistortion will be applied only to depth frames"),
                               gen.const("IR",       int_t, 3, "undistortion will be applied only to infrarred frames"),
                               gen.const("DEPTH_IR", int_t, 4, "undistortion will be applied to both depth and infra red frames")],
                             "List of possible values that can be passed to setUndistortionType")

frame_type = gen.enum([ gen.const("DEPTH_FRAME",    int_t, 1, "Depth Frame"),
                        gen.const("IR_FRAME",       int_t, 2, "Infrarred Frame"),
                        gen.const("DEPTH_IR_FRAME", int_t, 3, "Depth and Infrarred frame")],
                      "List of possible frames to be received")

gen.add("set_stream_fps", int_t, 0, "Sets camera stream Frames Per Second setting to fps", 30,  1, 30)
gen.add("set_stream_frame_type", int_t, 0, "Sets camera stream Frame Type setting to type", 1, 1, 3, edit_method=frame_type)
# gen.add("set_digital_output", int_t, 0, "", 0,  0, 1)
# gen.add("set_pwm", int_t, 0, "", 0,  0, 1024)
gen.add("set_undistortion_type", int_t, 0, "Enables undistortion operation", 2, 1, 4, edit_method=undistortion_type)
gen.add("set_gamma_correction", int_t, 0, "Sets gamma corection for IR", 0, 0, 2000)
gen.add("set_small_signal_threshold", int_t, 0, "Sets small signal removal", 0, 0, 2047)

exit(gen.generate(PACKAGE, "terabee", "Vga3DCamera"))
